code/utils/tests/test_conv.py

```diff
@@ -21,3 +21,17 @@ def test_conv_target_non_target():
   assert_almost_equal(actual_error_convolved, expected_error_convolved)
   assert_almost_equal(actual_target_convolved, expected_target_convolved)
   assert_almost_equal(actual_nontarget_convolved, expected_nontarget_convolved)
+
+def test_std():
+  task_fname = os.path.join(os.path.dirname(__file__), "test_data", "test_cond003.txt")
+  actual_std = conv.conv_std(10, task_fname, 1)
+  expected_std = np.array([ 0., -0.00030452, -0.00358456, -0.01000677, -0.01545791,-0.01714284, -0.01517165, -0.01111583, -0.00658934, -0.00265908])
+  assert_almost_equal(actual_std, expected_std )
+  
+
+def test_main():
+  task_fname = os.path.join(os.path.dirname(__file__), "test_data", "test_cond003.txt")
+  actual_main = conv_main(10, task_fname, 1)
+  expected_main = #I cannot run conv_main. There is something wrong with length.
```

Comment:
1. change line 79 at conv.py from neural_prediction = events2neural(filename, 0.1, n_trs) to neural_prediction = events2neural(filename, 0.1, n_trs, TR)

2. change line 40 at conv.py from target_neural, nontarget_neural, error_neural = events2neural_target_non_target(filename, error_fname, n_trs, tr_divs) to target_neural, nontarget_neural, error_neural = events2neural_target_non_target(filename, error_fname, n_trs, tr_divs, TR)

3. change line 34 test_conv.py actual_main = conv_main(10, task_fname, 1) to actual_main = conv_main(10, task_fname, 2.5)